javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/ReferenceMatcher.java

```diff
@@ -147,7 +157,64 @@ private boolean doesMatch(ClassLoader loader) {
     }
   }
 
-  public static List<Reference.Mismatch> checkMatch(
+  // for helper classes we make sure that all abstract methods from super classes and interfaces are
+  // implemented
+  private List<Reference.Mismatch> checkHelperClassMatch(Reference helperClass, TypePool typePool) {
+    List<Mismatch> mismatches = Collections.emptyList();
+
+    HelperReferenceWrapper helperWrapper = new Factory(typePool, references).create(helperClass);
+
+    if (!helperWrapper.hasSuperTypes() || helperWrapper.isAbstract()) {
```

Comment:
Do we support abstract helpers?

Reply:
We do -- we just skip the abstract method check for them, because it's perfectly fine for them to contain abstract methods.